### PR TITLE
ScriptCanvas cmake target cleanup and adjustment to support custom data autogen

### PR DESCRIPTION
**Pull request by onecent1101**
## What does this PR do?
Clean up unused cmake target and refactor header files so it can be used by other gems with light overhead.

## How was this PR tested?
No functional change, existing tests pass still.

Signed-off-by: onecent1101 <liug@amazon.com>

### CODE REVIEW

**Review by gadams3 — APPROVED** [review]

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Data/Constants.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#include <AzCore/Component/EntityId.h>
+
+namespace ScriptCanvas
+{
+    // #scriptcanvas_component_extension
+    // A place holder identifier for the AZ::Entity that owns the graph.
+    // The actual value in each location initialized to GraphOwnerId is populated with the owning entity at editor-time, Asset
+    // Processor-time, or runtime, as soon as the owning entity is known.
+    using GraphOwnerIdType = AZ::EntityId;
+    static const GraphOwnerIdType GraphOwnerId = AZ::EntityId(0xacedc0de);
```
Comment (pollend) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/Data/Constants.h:20]: can you use a constexpr here?

**Review by pollend — APPROVED** [review]

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Data/Constants.h (hunk 1 above)
Comment (onecent1101) [reply Gems/ScriptCanvas/Code/Include/ScriptCanvas/Data/Constants.h:20]: I believe AZ::EntityId we use here doesn't support compile time constant value.

**Review by lsemp3d — APPROVED** [review]
Very nice cleanup!

File: Gems/ScriptCanvas/Code/scriptcanvasgem_headers.cmake [hunk 2]
```diff
@@ -377,6 +371,11 @@ set(FILES
     Include/ScriptCanvas/Serialization/BehaviorContextObjectSerializer.h
     Include/ScriptCanvas/Serialization/DatumSerializer.h
     Include/ScriptCanvas/Serialization/RuntimeVariableSerializer.h
+    Include/ScriptCanvas/Data/BehaviorContextObject.h
+    Include/ScriptCanvas/Data/BehaviorContextObjectPtr.h
+    Include/ScriptCanvas/Data/Constants.h
+    Include/ScriptCanvas/Data/Data.h
+    Include/ScriptCanvas/Data/DataRegistry.h
```
Comment (lumberyard-employee-dm) [inline Gems/ScriptCanvas/Code/scriptcanvasgem_headers.cmake:378]: Should the other `Include/ScriptCanvas/Data` includes here be moved upwards instead?

File: Gems/ScriptCanvas/Code/scriptcanvasgem_common_files.cmake [hunk 3]
```diff
@@ -204,6 +200,10 @@ set(FILES
     Include/ScriptCanvas/Serialization/BehaviorContextObjectSerializer.cpp
     Include/ScriptCanvas/Serialization/DatumSerializer.cpp
     Include/ScriptCanvas/Serialization/RuntimeVariableSerializer.cpp
+    Include/ScriptCanvas/Data/BehaviorContextObject.cpp
+    Include/ScriptCanvas/Data/BehaviorContextObjectPtr.cpp
+    Include/ScriptCanvas/Data/Data.cpp
+    Include/ScriptCanvas/Data/DataRegistry.cpp
     Include/ScriptCanvas/Data/DataTrait.cpp
     Include/ScriptCanvas/Data/PropertyTraits.cpp
```
Comment (lumberyard-employee-dm) [inline Gems/ScriptCanvas/Code/scriptcanvasgem_common_files.cmake:208]: Repeat: These include paths should be moved upwards to maintain alphabetical order

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Core/ContainerTypeReflection.h [hunk 4]
```diff
@@ -5,12 +5,10 @@
...
  */
-#include <ScriptCanvas/Libraries/Libraries.h>
-#include <Data/DataMacros.h>
-#include <Data/DataTrait.h>
-#include <ScriptCanvas/Core/Attributes.h>
 
-#include <AzFramework/Entity/EntityContextBus.h>
+#include <AzCore/Script/ScriptContextAttributes.h>
+#include <ScriptCanvas/Data/DataMacros.h>
+#include <ScriptCanvas/Data/DataTraitBase.h>
```
Comment (lumberyard-employee-dm) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Core/ContainerTypeReflection.h:11]: Can you move this below the `#pragma once`

**Review by lumberyard-employee-dm — APPROVED** [review]

**Pull request merged**